synapse/groups/groups_server.py

```diff
@@ -615,6 +672,7 @@ async def invite_to_group(self, group_id, user_id, requester_user_id, content):
 
         if self.hs.is_mine_id(user_id):
             groups_local = self.hs.get_groups_local_handler()
+            assert isinstance(groups_local, GroupsLocalHandler)
```

Comment:
I think having a 500 error from the assertion is fine, but I'll add a message to it.

Reply:
Hmm, I'm not sure if "Misconfiguration of workers for groups." gives a sysadmin much to go on in terms of resolving the issue. Might still be nice to add a log line pointing out the issue? Or are we trying to keep things vague in case worker setup changes in the future?